ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -191,20 +191,23 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
   /// - SeeAlso:
   ///   - addKeyboard()
   /// - Returns: Whether the keyboard was set successfully
+  //TODO: this method appears unused, should we remove it?
   public func setKeyboard(withFullID fullID: FullKeyboardID) -> Bool {
-    if let keyboard = Storage.active.userDefaults.userKeyboard(withFullID: fullID) {
-      return setKeyboard(keyboard)
+    if let keyboard = Storage.active.userDefaults.userKeyboard(withFullID: fullID),
+       let _ = try? setKeyboard(keyboard) {
+        return true
     }
     return false
   }
+  
 
   /// Set the current keyboard.
   ///
-  /// - Returns: Whether the keyboard was set successfully
-  public func setKeyboard(_ kb: InstallableKeyboard) -> Bool {
+  /// - Throws: error if the keyboard was unchanged
+  public func setKeyboard(_ kb: InstallableKeyboard) throws {
```

Comment:
Is this a public API? If so, we can't change the behaviour.

It seems sensible to allow a no-op `setKeyboard`? Is this just to resolve unused result warnings?